src/transformer_util.ts

```diff
@@ -0,0 +1,545 @@
+      callExpr.arguments.length !== 1) {
+    return null;
+  }
+  const moduleExpr = callExpr.arguments[0];
+  if (moduleExpr.kind !== ts.SyntaxKind.StringLiteral) {
+    return null;
+  }
+  return (moduleExpr as ts.StringLiteral).text;
+}
+
+function lastNodeWith(nodes: ts.Node[], predicate: (node: ts.Node) => boolean): ts.Node|null {
+  for (let i = nodes.length - 1; i >= 0; i--) {
+    const node = nodes[i];
+    if (predicate(node)) {
+      return node;
+    }
+  }
+  return null;
+}
+
+/**
+ * Synthesizes the comments before and after a node
+ * befor calling a callback.
+ */
+export function visitNodeWithSynthesizedComments<T extends ts.Node>(
+    context: ts.TransformationContext, sourceFile: ts.SourceFile, node: T,
+    visitor: (node: T) => T): T {
+  if (node.flags & ts.NodeFlags.Synthesized) {
+    return visitor(node);
+  }
+  if (node.kind === ts.SyntaxKind.Block) {
+    const block = node as ts.Node as ts.Block;
+    node = visitNodeStatementsWithSynthesizedComments(
+        context, sourceFile, node, block.statements,
+        (node, stmts) => visitor(ts.updateBlock(block, stmts) as ts.Node as T));
+  } else if (node.kind === ts.SyntaxKind.SourceFile) {
+    node = visitNodeStatementsWithSynthesizedComments(
+        context, sourceFile, node, sourceFile.statements,
+        (node, stmts) => visitor(updateSourceFileNode(sourceFile, stmts) as ts.Node as T));
+  } else {
+    const fileContext = assertFileContext(context, sourceFile);
+    const leadingLastCommentEnd =
+        synthesizeLeadingComments(sourceFile, node, fileContext.lastCommentEnd);
+    const trailingLastCommentEnd = synthesizeTrailingComments(sourceFile, node);
+    if (leadingLastCommentEnd !== -1) {
+      fileContext.lastCommentEnd = leadingLastCommentEnd;
+    }
+    node = visitor(node);
+    if (trailingLastCommentEnd !== -1) {
+      fileContext.lastCommentEnd = trailingLastCommentEnd;
+    }
+  }
+  return resetNodeTextRangeToPreventDuplicateComments(node);
+}
+
+function resetNodeTextRangeToPreventDuplicateComments<T extends ts.Node>(node: T): T {
+  ts.setEmitFlags(node, (ts.getEmitFlags(node) || 0) | ts.EmitFlags.NoComments);
+  // Reset the text range for some special nodes as otherwise TypeScript
+  // would always emit the original comments for them.
+  // See also addSyntheticCommentsAfterTsTransformer.
+  let allowTextRange = node.kind !== ts.SyntaxKind.ClassDeclaration &&
+      node.kind !== ts.SyntaxKind.ExportDeclaration &&
+      node.kind !== ts.SyntaxKind.ImportDeclaration &&
+      node.kind !== ts.SyntaxKind.VariableDeclaration &&
+      !(node.kind === ts.SyntaxKind.VariableStatement &&
+        tsickle.hasModifierFlag(node, ts.ModifierFlags.Export));
+  if (node.kind === ts.SyntaxKind.PropertyDeclaration) {
+    allowTextRange = false;
+    const pd = node as ts.Node as ts.PropertyDeclaration;
+    // TODO(tbosch): Using pd.initializer! as the typescript typings for intializer are incorrect.
```

Comment:
I think I will leave `TODO`s everywhere for now where I need to file a bug, as that takes a bit (with small reproduction, ...)